File: Frontend/src/app/Models/post.ts
export class Post {
  constructor(
    public id: number,
    public userId: number,
    public content: string,
    public imageUrl?: string,
    public creationDate?: string,
    public categoryId?: number,
    public categoryName?: string,
    public userName?: string,
    public userProfile?: string
  ) {}
}

File: Backend/CybageConnect.Entity/Models/Post.cs
using System;
using System.Collections.Generic;

namespace CybageConnect.Entity.Models;

public partial class Post
{
    public int Id { get; set; }

    public int UserId { get; set; }

    public string Content { get; set; } = null!;

    public string? ImageUrl { get; set; }

    public DateTime CreationDate { get; set; }

    public int? CategoryId { get; set; }

    public virtual Category? Category { get; set; }

    public virtual ICollection<Comment> Comments { get; set; } = new List<Comment>();

    public virtual ICollection<Like> Likes { get; set; } = new List<Like>();

    public virtual User User { get; set; } = null!;
}
